src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -536,13 +549,28 @@ def resolve_imports_and_packages(
     source_code_display: bool = False,
     skip_upload_on_content_match: bool = False,
     force_push: bool = True,
-) -> Tuple[str, str, str, str, str]:
+) -> Tuple[str, str, str, str, str, bool]:
     upload_stage = (
         unwrap_stage_location_single_quote(stage_location)
         if stage_location
         else session.get_session_stage()
     )
 
+    # resolve packages
```

Comment:
Packages should be resolved before imports because package resolution can lead to imports (let's say, if local files or directories or specified in the requirements.txt file)